generator/Data/XCB/Python/Parse.hs

```diff
@@ -495,26 +590,48 @@ mkStructStyleUnpack prefix ext m membs =
         put $ st { stNames = stNames st ++ maybeToList name
                  , stPacks = packs
                  }
-        mkUnpackStmtsR xs
+        mkUnpackStmts xs
 
-      mkUnpackStmtsR (Right (listName, list, cons, listSz) : xs) = do
+      mkUnpackStmts (Right (listName, lists, listSz) : xs) = do
         (packNames, packStmt, packSz) <- flushAcc
         st <- get
         put $ st { stNeedsPad = True }
-        let pad = if stNeedsPad st
-                  then [typePad cons]
-                  else []
-        (restNames, restStmts, restSz) <- mkUnpackStmtsR xs
+        let listStmts = case lists of
+                          [] -> error "List should not be empty"
```

Comment:
For a lot of these you could use a pattern guard to kill the monad instead of these cases, so:

```
let ...
guard $ null lists
listStmts = mkList listName list (stNeedsPad st) cons
```